api/python/provisioner/commands/validator/validator.py

```diff
@@ -255,3 +259,72 @@ class YumRepoDataValidator(DirValidator):
     def __init__(self):
         super().__init__({Path("repomd.xml"): FileValidator(required=True)},
                          required=True)
+
+
+@attr.s
+class HashSumValidator(PathValidator):
+
+    """
+    Validator of hash-sum for the provided file and expected hash-sum for this
+    file.
+
+    Attributes
+    ----------
+    hash_sum: Union[str, bytes, bytearray]
+        Hexadecimal string or byte-array object with expected hash-sum value
+        of validated file.
+    hash_type: HashType
+        Type of hash sum. See `Hashtype` for more information
+
+    """
+    _hash_sum: Union[str, bytes, bytearray] = attr.ib(
+        validator=attr.validators.instance_of((str, bytes, bytearray))
+    )
+    _hash_type: HashType = attr.ib(
+        validator=attr.validators.in_(HashType),
+        default=HashType.MD5,
+        converter=lambda x: HashType.MD5 if x is None else HashType(x)
+    )
+
+    def validate(self, path: Path):
+        """
+        Validates if hash-sum of the file provided by `path` matches
+        the attribute value of `_hash_sum`.
+
+        Parameters
+        ----------
+        path: Path
+            path to the file which hash-sum will be validated
+
+        Returns
+        -------
+        None
+
+        Raises
+        ------
+        ValidationError
+            If validation is failed.
+        """
+        file_validator = FileValidator(required=True)
+        file_validator.validate(path)
+
+        if self._hash_type.value not in hashlib.algorithms_available:
+            raise ValidationError(f"Hash type '{self._hash_type.value}' is not"
```

Comment:
I did this for one reason, 
https://docs.python.org/3/library/hashlib.html

>Constructors for hash algorithms that are always present in this module are sha1(), sha224(), sha256(), sha384(), sha512(), blake2b(), and blake2s(). md5() is normally available as well, though it may be missing or blocked if you are using a rare “FIPS compliant” build of Python. Additional algorithms may also be available depending upon the OpenSSL library that Python uses on your platform. On most platforms the sha3_224(), sha3_256(), sha3_384(), sha3_512(), shake_128(), shake_256() are also available.

Depending on configuration of OpenSSL some algorithms can be missed. OpenSSL lib can be deleted or broken. So this validation is simple and doesn't require a big resources, but will helpful in case of broken system.